### ISSUE TITLE
Adjust generated scenarios

### ISSUE THREAD

**Issue by Turbo87** [ember-cli/ember-try-config#13]
As mentioned in https://github.com/ember-cli/ember-try-config/pull/3 there are some situations where it would be useful to adjust the list of generated scenarios afterwards by e.g. adding or overwriting certain dependencies or even cloning generated scenarios.

Instead of implicitly generating the scenarios when `useVersionCompatibility` is used I would suggest to explicitly `require('ember-try-config')` and then put something like `scenarios: generateScenarios()` in your `config/ember-try.js` file.

That way the `generateScenarios()` result could be written into a temporary variable if needed and then adjusted afterwards.

/cc @rwjblue @kategengler

**Comment by rwjblue** [comment]
Agreed.

**Comment by Turbo87** [comment]
@rwjblue I just noticed an issue with this proposal which unfortunately makes it a little harder to backport: the `generateScenarios()` function would have to be async and I don't think `ember-try` is able to handle async yet... 🤔

**Comment by kategengler** [comment]
That seems fine to me, so long as the use of `versionCompatibility` in the package.json stays (I still have an outside hope of using that to get addons to declare their compatibility).

**Comment by Turbo87** [comment]
@kategengler yes, I'm fine with that. the default of `generateScenarios()` would be to look at the `package.json` file for `versionCompatibility` unless specified otherwise.

**Comment by locks** [comment]
Is this still relevant?

**Comment by kategengler** [comment]
Yes, discussion of a similar solution is ongoing, including here: https://github.com/ember-cli/ember-try-config/pull/192#issuecomment-1677690162
